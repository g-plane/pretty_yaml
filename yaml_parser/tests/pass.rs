use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
