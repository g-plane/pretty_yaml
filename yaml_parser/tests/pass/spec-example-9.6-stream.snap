---
source: yaml_parser/tests/pass.rs
---
ROOT@0..53
  DOCUMENT@0..8
    FLOW@0..8
      PLAIN_SCALAR@0..8 "Document"
  WHITESPACE@8..9 "\n"
  DOCUMENT@9..24
    DIRECTIVES_END@9..12 "---"
    WHITESPACE@12..13 "\n"
    COMMENT@13..20 "# Empty"
    WHITESPACE@20..21 "\n"
    DOCUMENT_END@21..24 "..."
  WHITESPACE@24..25 "\n"
  DOCUMENT@25..52
    DIRECTIVE@25..34
      PERCENT@25..26 "%"
      YAML_DIRECTIVE@26..34
        DIRECTIVE_NAME@26..30 "YAML"
        WHITESPACE@30..31 " "
        YAML_VERSION@31..34 "1.2"
    WHITESPACE@34..35 "\n"
    DIRECTIVES_END@35..38 "---"
    WHITESPACE@38..39 "\n"
    BLOCK@39..52
      BLOCK_MAP@39..52
        BLOCK_MAP_ENTRY@39..52
          BLOCK_MAP_KEY@39..48
            FLOW@39..48
              PLAIN_SCALAR@39..48 "matches %"
          COLON@48..49 ":"
          WHITESPACE@49..50 " "
          BLOCK_MAP_VALUE@50..52
            FLOW@50..52
              PLAIN_SCALAR@50..52 "20"
  WHITESPACE@52..53 "\n"
